native/src/main/java/io/ballerina/lib/data/jsondata/json/JsonParser.java

```diff
@@ -249,7 +249,7 @@ public Object execute(Reader reader, BMap<BString, Object> options, Type type) t
                 }
                 case TypeTags.INTERSECTION_TAG -> {
                     Type effectiveType = ((IntersectionType) type).getEffectiveType();
-                    if (!SymbolFlags.isFlagOn(SymbolFlags.READONLY, effectiveType.getFlags())) {
```

Comment:
Shouldn't this have worked too?

Reply:
I think its a bug in the ballerina lang.
Will check and create a issue

Reply:
Created. https://github.com/ballerina-platform/ballerina-lang/issues/43484